### PR TITLE
WPCOM: Migrate `wpcom.undocumented().site()` to `wpcom.req`

### PR DESCRIPTION
**Pull request by tyxla**
#### Changes proposed in this Pull Request

Currently, all  `wpcom.undocumented().site()` methods are used only from within the `requestSiteStats()` thunk. 

This PR migrates all the usages to `wpcom.req.get()` and removes the obsolete methods with their constructor and prototype.

Part of the ongoing effort to get rid of `wpcom.undocumented()`.

#### Testing instructions

* While testing the following, keep an eye on HTTP requests and verify all are still loading well and there are no 404s.
* Compare the following Calypso screens on this branch and trunk, and verify they work the same way a WP.com simple site, an Atomic site, and a site that has WooCommerce installed with some orders in it:
  * `/home/:site`
  * `/stats/day/:site`
  * other Stats pages - navigate to different Stats screens from the previous one.
  * `/store/stats/orders/day/:site` - only for a store with WooCommerce installed 
* Verify tests still pass: `yarn run test-client client/state/stats/lists/test/actions.js`

### CODE REVIEW

**Comment by tyxla** [comment]
Rebased since we recently merged a bunch of undocumented removals.

**Review by lynnbgriffith — APPROVED** [review]
👍 This tests well for me! Thanks!

File: client/state/stats/lists/actions.js [hunk 1]
```diff
@@ -47,23 +61,29 @@ export function requestSiteStats( siteId, statType, query ) {
...
-				'statsStoreReferrers',
-			],
-			statType
-		);
+
+		let subpath;
+		let apiNamespace;
+		if ( wpcomV1Endpoints.hasOwnProperty( statType ) ) {
+			subpath = wpcomV1Endpoints[ statType ];
+			apiNamespace = 'rest/v1.1';
```
Comment (jsnajdr) [inline client/state/stats/lists/actions.js:69]: You could also leave `apiNamespace` undefined here.

File: client/lib/wpcom-undocumented/lib/site.js [hunk 2]
```diff
@@ -1,179 +0,0 @@
-import debugFactory from 'debug';
-const debug = debugFactory( 'calypso:wpcom-undocumented:site' );
-
-/**
- * Resources array
- * A list of endpoints with the same structure
- * [  wpcom-undocumented.functionName, siteAPiSubPath, apiVersion ]
- */
-const resources = [
-	[ 'statsEvents', 'posts/' ],
```
Comment (jsnajdr) [inline client/lib/wpcom-undocumented/lib/site.js:10]: The `statsEvents` one was never used?

**Review by jsnajdr — COMMENTED** [review]
Thanks for working on this one! I was always too intimidated by this code to even try 🙂

File: client/state/stats/lists/actions.js (hunk 1 above)
Comment (tyxla) [reply client/state/stats/lists/actions.js:69]: Yeah, I just decided to keep it for clarity and a better understanding of what's going on.

File: client/lib/wpcom-undocumented/lib/site.js (hunk 2 above)
Comment (tyxla) [reply client/lib/wpcom-undocumented/lib/site.js:10]: Good question - yup, I verified it wasn't in use anymore and removed it.

File: client/state/stats/lists/actions.js (hunk 1 above)
Comment (tyxla) [reply client/state/stats/lists/actions.js:69]: But you were right of course! This broke something - see https://github.com/Automattic/wp-calypso/pull/57814 and https://github.com/Automattic/wp-calypso/issues/57775

**Pull request merged**